src/cryptography/x509/name.py

```diff
@@ -54,6 +54,10 @@ def __repr__(self):
 
 class Name(object):
     def __init__(self, attributes):
+        attributes = list(attributes)
+        if not all(isinstance(x, NameAttribute) for x in attributes):
+            raise TypeError("attributes must be a list of NameAttribute")
```

Comment:
Behavior change: Name now checks that attributes elements are of type NameAttribute